typing/parmatch.ml

```diff
@@ -897,7 +897,7 @@ let pat_of_constr ex_pat cstr =
 let orify x y = make_pat (Tpat_or (x, y, None)) x.pat_type x.pat_env
 
 let rec orify_many = function
-| [] -> assert false
+| [] -> raise Empty
```

Comment:
Looking back at this code, I'm concerned by this exception.
`orify_many` is called in many places, and are you sure that they are all ready to catch `Empty`?
My feeling is that you should rather modify `ppat_of_type` to raise an exception when `pats = []`, without calling `orify_many` on the empty list. And this should be documented in the .mli.